FWCore/Framework/interface/limited/OutputModuleBase.h

```diff
@@ -131,34 +130,23 @@ namespace edm {
 
       void doBeginStream(StreamID id);
       void doEndStream(StreamID id);
-      void doStreamBeginRun(StreamID id, RunPrincipal& ep, EventSetupImpl const& c, ModuleCallingContext const*);
-      void doStreamEndRun(StreamID id, RunPrincipal& ep, EventSetupImpl const& c, ModuleCallingContext const*);
-      void doStreamBeginLuminosityBlock(StreamID id,
-                                        LuminosityBlockPrincipal& ep,
-                                        EventSetupImpl const& c,
-                                        ModuleCallingContext const*);
-      void doStreamEndLuminosityBlock(StreamID id,
-                                      LuminosityBlockPrincipal& ep,
-                                      EventSetupImpl const& c,
-                                      ModuleCallingContext const*);
-
-      bool doEvent(EventPrincipal const& ep, EventSetupImpl const& c, ActivityRegistry*, ModuleCallingContext const*);
+      void doStreamBeginRun(StreamID, RunTransitionInfo&, ModuleCallingContext const*);
```

Comment:
Why is the `RunTransitionInfo&` not `const`? It is for `doBeginRun`?

Reply:
I was copying what was there before. If the RunPrincipal was const, then I made the RunTransitionInfo const. In general, I was doing this in many places in this PR without thinking further.

On closer examination, these four functions here and also four similar functions in global/OutputModuleBase.h are not implemented and not needed for anything. The has_stream_functions template is not specialized for these OutputModuleBase types so these functions are never called. This is just dead code. I will push a commit soon to just delete them.

https://cmssdt.cern.ch/lxr/source/FWCore/Framework/src/WorkerT.cc#0030